cloud-platform/app-deploy.md

```diff
@@ -0,0 +1,212 @@
+
+
+### Service
+
+Service files are used to specify port and protocol information for your application and are also used to bundle together the set of pods created by the deployment.
+
+See the contents of the `service.yml` file below:
+
+```Yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: cp-app-demo-svc
+  labels:
+    app: cp-app-demo-svc
+spec:
+  ports:
+  - port: 80
+    name: http
+    targetPort: 80
+  selector:
+    app: cp-demo-app
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app-svc` tags with ones suited to your application.
+
+Also, ensure the `selector:` app tag is the same as specified in the `deployment.yml`.
+
+*Tip:* You can find more info on service definition in the [kubernetes docs](https://kubernetes.io/docs/tasks/access-application-cluster/service-access-application-cluster/).
+
+### Ingress
+
+Ingress files are to use to define external access to the application.
+
+See the contents of the `ingress.yml` file below:
+
+```Yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: cp-demo-app-ing
+spec:
+  rules:
+  - host: cp-demo-app.integration.dsd.io
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: cp-demo-app-svc
+          servicePort: 80
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app-ing` tag with one suited to your application.
+
+Also, ensure the `serviceName:` tag is the same as specified in the `service.yml`.
+
+*Tip:* You can find more info on ingress in the [kubernetes docs](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+
+## Deploying application to the cluster
+
+With all of the deployment files configured, you can now deploy your application to the Cloud Platform.
+
+Start by listing the namespaces on the cluster you are connected to:
+
+`kubectl get namespaces`
+
+The list that gets returned should include the environment you [created earlier](/cloud-platform/env-create), here we assume it is called `my-app-dev`.
+
+To deploy your application run the following command, that points to the directory where the deployment files are stored, in the [reference app]() they are in the `kubectl_deploy` directory:
+
+`kubectl create -f kubectl_deploy --namespace my-app-dev`
+
+You have to specify the namespace you want to deploy to, this should be the namespace of the environment you created.
+
+Confirm the deployment with:
+
+`kubectl get pods --namespace my-app-dev`
+
+## Interacting with the application
+
+With the application deployed into the Cloud Platform, there are a few ways of managing it:
+
+* **View pods** - `kubectl get pods --namespace my-app-dev`
+* **Check host** - `kubectl get ing --namespace my-app-dev`
+* **Delete application** - `kubectl delete -f deployment-files --namespace my-app-dev`
+* **Shell into container** - `kubectl exec -it pod-name -- /bin/bash --namespace my-app-dev`
```

Comment:
That'll error out I think - the command you're running in the container has to be at the end, so:

`kubectl exec -it --namespace my-app-dev pod-name -- /bin/bash`

Reply:
Also the long form of the `-it` flags might be a bit more informative?

`kubectl exec --stdin --tty --namespace my-app-dev pod-name -- /bin/bash`

Reply:
Yes you are right 👍🏽